kernel/src/main/scala/cats/kernel/Order.scala

```diff
@@ -0,0 +1,209 @@
+
+/**
+ * The `Order` type class is used to define a total ordering on some type `A`.
+ * An order is defined by a relation <=, which obeys the following laws:
+ *
+ * - either x <= y or y <= x (totality)
+ * - if x <= y and y <= x, then x == y (antisymmetry)
+ * - if x <= y and y <= z, then x <= z (transitivity)
+ *
+ * The truth table for compare is defined as follows:
+ *
+ * x <= y    x >= y      Int
+ * true      true        = 0     (corresponds to x == y)
+ * true      false       < 0     (corresponds to x < y)
+ * false     true        > 0     (corresponds to x > y)
+ *
+ * By the totality law, x <= y and y <= x cannot be both false.
+ */
+trait Order[@sp A] extends Any with PartialOrder[A] { self =>
+
+  /**
+   * Result of comparing `x` with `y`. Returns an Int whose sign is:
+   * - negative iff `x < y`
+   * - zero     iff `x = y`
+   * - positive iff `x > y`
+   */
+  def compare(x: A, y: A): Int
+
+  def partialCompare(x: A, y: A): Double = compare(x, y).toDouble
+
+  /**
+   * If x <= y, return x, else return y.
+   */
+  def min(x: A, y: A): A = if (lt(x, y)) x else y
+
+  /**
+   * If x >= y, return x, else return y.
+   */
+  def max(x: A, y: A): A = if (gt(x, y)) x else y
+
+  /**
+   * Defines an order on `B` by mapping `B` to `A` using `f` and using `A`s
+   * order to order `B`.
+   */
+  override def on[@sp B](f: B => A): Order[B] =
+    new Order[B] {
+      def compare(x: B, y: B): Int = self.compare(f(x), f(y))
+    }
+
+  /**
+   * Defines an ordering on `A` where all arrows switch direction.
+   */
+  override def reverse: Order[A] =
+    new Order[A] {
+      def compare(x: A, y: A): Int = self.compare(y, x)
+
+      override def reverse: Order[A] = self
+    }
+
+  // The following may be overridden for performance:
+
+  /**
+   * Returns true if `x` = `y`, false otherwise.
+   */
+  override def eqv(x: A, y: A): Boolean =
+    compare(x, y) == 0
+
+  /**
+   * Returns true if `x` != `y`, false otherwise.
+   */
+  override def neqv(x: A, y: A): Boolean =
+    compare(x, y) != 0
+
+  /**
+   * Returns true if `x` <= `y`, false otherwise.
+   */
+  override def lteqv(x: A, y: A): Boolean =
+    compare(x, y) <= 0
+
+  /**
+   * Returns true if `x` < `y`, false otherwise.
+   */
+  override def lt(x: A, y: A): Boolean =
+    compare(x, y) < 0
+
+  /**
+   * Returns true if `x` >= `y`, false otherwise.
+   */
+  override def gteqv(x: A, y: A): Boolean =
+    compare(x, y) >= 0
+
+  /**
+   * Returns true if `x` > `y`, false otherwise.
+   */
+  override def gt(x: A, y: A): Boolean =
+    compare(x, y) > 0
+
+  /**
+   * Returns a new `Order[A]` instance that first compares by the original
+   * `Order` instance and uses the provided `Order` instance to "break ties".
+   *
+   * That is, `x.whenEqual(y)` creates an `Order` that first orders by `x` and
+   * then (if two elements are equal) falls back to `y` for the comparison.
+   */
+  def whenEqual(o: Order[A]): Order[A] = new Order[A] {
```

Comment:
out of curiosity, does this make a bounded semilattice like the equals ones does? Seems like, with `empty` being `allEqual`? I guess it is a Band, not a Semilattice since it is not commutative.